packages/php/email-editor/src/Engine/class-email-editor.php

```diff
@@ -267,4 +268,28 @@ public function extend_email_theme_styles( WP_Theme_JSON $theme, WP_Post $post )
 		}
 		return $theme;
 	}
+
+	/**
+	 * Load JS vars
+	 *
+	 * @return void
+	 * @throws \InvalidArgumentException If the post-type is invalid.
+	 */
+	public function load_js_vars() {
+		global $post;
+
+		$email_editor_post_type_names      = array_column( $this->get_post_types(), 'name' );
+		$email_editor_current_post_type    = $post->post_type;
+		$current_post_is_email_editor_type = in_array( $email_editor_current_post_type, $email_editor_post_type_names, true );
+
+		if ( ! $current_post_is_email_editor_type ) {
+			throw new \InvalidArgumentException( esc_html__( 'Invalid email post type', 'mailpoet' ) );
+		}
+
+		?>
+		<script type="text/javascript"> <?php // phpcs:ignore ?>
+			window.mailpoet_email_editor_current_post_type = <?php echo wp_json_encode( $email_editor_current_post_type ); ?>;
+		</script>
+		<?php
+	}
```

Comment:
Just a thought. We currently ask the integrator to provide a couple of [variables that are essential for the editor](https://github.com/mailpoet/mailpoet/blob/3d45ea92e2097c9211e735830f9ac667ae7d1393/mailpoet/lib/EmailEditor/Integrations/MailPoet/EditorPageRenderer.php#L151-L153). I feel that `mailpoet_email_editor_current_post_type` is very similar in that regard. It would be nicer if the editor package could prepare more itself. But, as we already have some variables there, I think that as for now we could add the current post type among them. Perhaps later, when we focus on simplifying bootstrapping the editor, we could move that somehow to the package. I'm suggesting that for the sake of having them all in one place.

Reply:
Hmm, Okay. Done in https://github.com/mailpoet/mailpoet/pull/6077/commits/12344da42842a10c6e08c007e0e1e414d8d4241d